browser/src/VariantList/mergeExomeAndGenomeData.ts

```diff
@@ -1,29 +1,61 @@
+import { DatasetId } from '@gnomad/dataset-metadata/metadata'
 import { Filter } from '../QCFilter'
 import { Population, Variant } from '../VariantPage/VariantPage'
+import { PopulationId, getPopulationsInDataset } from '@gnomad/dataset-metadata/gnomadPopulations'
 
 // safe math on possibly null values
 const add = (n1: number | null | undefined, n2: number | null | undefined) => (n1 || 0) + (n2 || 0)
 
+// include placeholders for any ancestries missing from the dataset
```

Comment:
This looks like it should work but as a rule it's good to avoid approaches using mutable data unless using immutable data makes the algorithm much more complex or less efficient. (Don't want to bloat up this comment more by reviewing things you may already have thought about, but if the rationale for why this is so isn't clear I'm happy to say more about it).

Now assuming that the ordering of the ancestries in the output doesn't matter, one thing you could do is something like:

```
const fullAncestries = versionAncestries.flatMap((versionAncestry) =>
 findAncestries(fullAncestries, versionAncestry) || emptyAncestries
)
```

...where you have `emptyAncestry` and `findAncestry` defined appropriately.